servlet.py

```diff
@@ -561,6 +565,7 @@ def get(self):
                         self.send_error(400, explanation='Invalid file type %s' % mtype)
         else:
             self.send_error(400, explanation='That pair is not installed')
+            sem.release()
```

Comment:
Instead of acquire and release so far away from each other, can you rewrite it to use the `with (yield sem.acquire()):` pattern? And just around the call to `translation.translateDoc`; does't need to be around the whole  `get` function.